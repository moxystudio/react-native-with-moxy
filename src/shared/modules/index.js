/* istanbul ignore file */

export { default as Button } from './react-native-button';
export { default as SplashScreen } from './react-native-splash-screen';
export { default as SafeAreaHeader } from './react-native-safe-area-header';
export * from './react-theme';
export * from './react-native-locale';
export * from './react-native-app-config';
