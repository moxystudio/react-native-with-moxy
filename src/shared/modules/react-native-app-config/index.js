import AppConfigContext from './context';

const AppConfigConsumer = AppConfigContext.Consumer;

export { default as AppConfigProvider } from './AppConfigProvider';
export { AppConfigConsumer };
export { default as useAppConfig } from './use-app-config';
export { default as withAppConfig } from './with-app-config';
