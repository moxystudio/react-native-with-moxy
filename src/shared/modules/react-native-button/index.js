/* istanbul ignore file */

export { default } from './Button';
