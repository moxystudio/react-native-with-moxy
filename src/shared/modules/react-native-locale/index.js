import LocaleContext from './context';

const LocaleConsumer = LocaleContext.Consumer;

export { default as LocaleProvider } from './LocaleProvider';
export { LocaleConsumer };
export { default as useLocale } from './use-locale';
export { default as withLocale } from './with-locale';
