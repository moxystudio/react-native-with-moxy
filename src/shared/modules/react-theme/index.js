/* istanbul ignore file */

export { default as ThemeProvider } from './ThemeProvider';
export { default as useTheme } from './useTheme';
