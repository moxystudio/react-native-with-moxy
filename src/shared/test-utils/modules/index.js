export { AppTree } from './react-app-tree';
