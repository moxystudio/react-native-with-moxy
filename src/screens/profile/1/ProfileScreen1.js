import React from 'react';
import { Text } from 'react-native';
import { useTheme } from '../../../shared/components';
import { SafeAreaView } from 'react-native-safe-area-context';

import buildStyles from '../styles';

const ProfileScreen1 = () => {
    const { themeStyles } = useTheme();
    const styles = buildStyles(themeStyles);

    return (
        <SafeAreaView style={ styles.container }>
            <Text>This is the ProfileScreen1</Text>
        </SafeAreaView>
    );
};

export default ProfileScreen1;
