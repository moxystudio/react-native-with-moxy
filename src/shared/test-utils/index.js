import React from 'react';
import { Text } from 'react-native';
import PropTypes from 'prop-types';
import { render } from '@testing-library/react-native';
import { SafeAreaProvider } from 'react-native-safe-area-context';
import idObj from 'identity-obj-proxy';
import { AppConfigProvider, ThemeProvider, LocaleProvider } from '../modules';

const localeConfig = {
    locales: [
        {
            id: 'en-US',
            name: 'English',
            messages: idObj,
        },
        {
            id: 'pt-PT',
            name: 'PortuguÃªs',
            messages: idObj,
        },
    ],
    // Consider the following React code:
    //
    // <Text>
    //    <FormattedMessage id="foo" />
    // </Text>
    //
    // In this case, the "foo" key will resolve into the text "bar", and the following will be rendered:
    //
    // <Text>
    //    <Text>
    //       bar
    //    </Text>
    // </Text>
    //
    // This a is necessary workaround because *ByText queries can only look up a string if it's
    // a direct child of a Text component.
    textComponent: Text,
};

const Providers = ({ children }) => (
    <AppConfigProvider>
        <SafeAreaProvider>
            <ThemeProvider>
                <LocaleProvider { ...localeConfig }>
                    { children }
                </LocaleProvider>
            </ThemeProvider>
        </SafeAreaProvider>
    </AppConfigProvider>
);

Providers.propTypes = {
    children: PropTypes.element.isRequired,
};

const customRender = (ui, options) => render(ui, { wrapper: Providers, ...options });

// Export everything from NTL
export * from '@testing-library/react-native';

// Override render method
export { customRender as render };
