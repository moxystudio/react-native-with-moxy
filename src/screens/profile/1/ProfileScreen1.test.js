import React from 'react';
import { mount } from 'enzyme';
import SafeAreaView from 'react-native-safe-area-view';
import { AppTree } from '../../../shared/test-utils/modules';
import ProfileScreen1 from './ProfileScreen1';

it('should render a safe area view', () => {
    const tree = mount(
        <AppTree>
            <ProfileScreen1 />
        </AppTree>,
    );

    expect(tree.find(SafeAreaView).exists()).toBe(true);
});

it('should render screen title', () => {
    const tree = mount(
        <AppTree>
            <ProfileScreen1 />
        </AppTree>,
    );

    const title = tree.find("[accessibilityLabel='title']").first().text();

    expect(title).toBe('profile1.title');
});
