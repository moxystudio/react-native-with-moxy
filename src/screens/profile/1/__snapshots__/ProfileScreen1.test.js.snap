// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly 1`] = `
<AppTree>
  <SafeAreaProvider>
    <ThemeContextProvider>
      <ProfileScreen1>
        <SafeAreaView
          style={
            Object {
              "alignItems": "center",
              "flex": 1,
              "justifyContent": "center",
            }
          }
        >
          <Text>
            <Text>
              This is the ProfileScreen1
            </Text>
          </Text>
        </SafeAreaView>
      </ProfileScreen1>
    </ThemeContextProvider>
  </SafeAreaProvider>
</AppTree>
`;
